fdbclient/ManagementAPI.actor.cpp

```diff
@@ -1046,7 +1046,6 @@ ACTOR Future<Optional<ClusterConnectionString>> getClusterConnectionStringFromSt
 			return Optional<ClusterConnectionString>();
 		}
 
-		Version readVersion = wait(tr->getReadVersion());
```

Comment:
An assumption here is that function calls don't have side-effects e.g. `tr->getReadVersion()` shouldn't mutate any state. If such functions are marked as const, then I think it's safe. So curious, for non-const functions, how do you ensure there are no side-effects? Do we rely on simulation testing or there's more due diligence required?

Reply:
For this specific case, the getReadVersion request will be sent to the GRV proxy anyway without this explicit call. I.e., if we removed this line, the next line `tr->get(coordinatorsKey)` will internally issue the same getReadVersion request , so there is no difference with or without this line.

Generally, for function calls we are not "quite" certain, simulation testing is unusually good at catching errors. So if simulation passes, we think the change is fine. I've run the simulation to confirm there are no failures w.r.t. this PR.